dev/release/02-source.sh

```diff
@@ -50,104 +60,118 @@ archive_name=tmp-apache-arrow
-
-# sign the archive
-gpg --armor --output ${tarball}.asc --detach-sig ${tarball}
-shasum -a 256 $tarball > ${tarball}.sha256
-shasum -a 512 $tarball > ${tarball}.sha512
-
-# check out the arrow RC folder
-svn co --depth=empty https://dist.apache.org/repos/dist/dev/arrow tmp
-
-# add the release candidate for the tag
-mkdir -p tmp/${tagrc}
-
-# copy the rc tarball into the tmp dir
-cp ${tarball}* tmp/${tagrc}
-
-# commit to svn
-svn add tmp/${tagrc}
-svn ci -m "Apache Arrow ${version} RC${rc}" tmp/${tagrc}
-
-# clean up
-rm -rf tmp
-
-rc_url="https://dist.apache.org/repos/dist/dev/arrow/${tagrc}"
-echo "Success! The release candidate is available here:"
-echo "  ${rc_url}"
-echo ""
-echo "Commit SHA1: ${release_hash}"
-echo ""
-echo "The following draft email has been created to send to the"
-echo "dev@arrow.apache.org mailing list"
-echo ""
-echo "---------------------------------------------------------"
-jira_url="https://issues.apache.org/jira"
-jql="project%20%3D%20ARROW%20AND%20status%20in%20%28Resolved%2C%20Closed%29%20AND%20fixVersion%20%3D%20${version}"
-n_resolved_issues=$(curl "${jira_url}/rest/api/2/search/?jql=${jql}" | jq ".total")
-cat <<MAIL
-To: dev@arrow.apache.org
-Subject: [VOTE] Release Apache Arrow ${version} - RC${rc}
-
-Hi,
-
-I would like to propose the following release candidate (RC${rc}) of Apache
-Arrow version ${version}. This is a release consiting of ${n_resolved_issues}
-resolved JIRA issues[1].
-
-This release candidate is based on commit:
-${release_hash} [2]
-
-The source release rc${rc} is hosted at [3].
-The binary artifacts are hosted at [4][5][6][7].
-The changelog is located at [8].
-
-Please download, verify checksums and signatures, run the unit tests,
-and vote on the release. See [9] for how to validate a release candidate.
-
-The vote will be open for at least 72 hours.
-
-[ ] +1 Release this as Apache Arrow ${version}
-[ ] +0
-[ ] -1 Do not release this as Apache Arrow ${version} because...
-
-[1]: ${jira_url}/issues/?jql=${jql}
-[2]: https://github.com/apache/arrow/tree/${release_hash}
-[3]: ${rc_url}
-[4]: https://bintray.com/apache/arrow/centos-rc/${version}-rc${rc}
-[5]: https://bintray.com/apache/arrow/debian-rc/${version}-rc${rc}
-[6]: https://bintray.com/apache/arrow/python-rc/${version}-rc${rc}
-[7]: https://bintray.com/apache/arrow/ubuntu-rc/${version}-rc${rc}
-[8]: https://github.com/apache/arrow/blob/${release_hash}/CHANGELOG.md
-[9]: https://cwiki.apache.org/confluence/display/ARROW/How+to+Verify+Release+Candidates
+# Create a dummy .git/ directory to download the source files from GitHub with Source Link in C#.
+mkdir ${tag}/.git && cd $_
+echo ${release_hash} > HEAD
+echo '[remote "origin"] url = https://github.com/apache/arrow.git' >> config
+mkdir objects refs
+cd -
+
+if [ ${PREPARE_RELEASE_TAG} -gt 0 ]; then
+  # Create new tarball from modified source directory
+  tar czf ${tarball} ${tag}
+  rm -rf ${tag}
+
+  ${SOURCE_DIR}/run-rat.sh ${tarball}
```

Comment:
How about move RAT check outside of `PREPARE_RELEASE_TAG`?
RAT check is useful.
How about adding `SOURCE_RAT`?

Reply:
I've added `SOURCE_RAT`.